templates/components/vectordbs/python/weaviate/vectordb.py

```diff
@@ -6,19 +6,40 @@
 DEFAULT_INDEX_NAME = "LlamaIndex"
 
 
-def _create_weaviate_client():
-    cluster_url = os.getenv("WEAVIATE_CLUSTER_URL")
-    api_key = os.getenv("WEAVIATE_API_KEY")
-    if not cluster_url or not api_key:
-        raise ValueError("Environment variables: WEAVIATE_CLUSTER_URL and WEAVIATE_API_KEY are required.")
-    auth_credentials = weaviate.auth.AuthApiKey(api_key)
-    client = weaviate.connect_to_weaviate_cloud(cluster_url, auth_credentials)
-    return client
+class WeaviateClientSingleton:
```

Comment:
Since the client is initialized by our app, managing the Weaviate connection using a singleton instance is sufficient for me.

Reply:
yes, you might even simplify it by putting the client in a global variable - same effect and probably easier to read for Python newbies